src/lib/utils/functions.test.ts

```diff
@@ -1,5 +1,128 @@
+    "expected_output": {
+      "nft": "Green",
+      "uri": ["Metadata stored in tokenUri (on-chain)", "Image is embedded in metadata"]
+    },
+  },
+}
+
+describe("isBase64", function () {
+
+  test('#isBase64', () => {
+    expect(isBase64("not base64")).toBe(false);
+
+    const expected_decoded = "Many hands make light work.";
+    const expected_encoded = "TWFueSBoYW5kcyBtYWtlIGxpZ2h0IHdvcmsu";
+    expect(isBase64(expected_encoded)).toBe(true);
+  });
+
+  test('#isBase64#devs for revolution uri', () => {
+    const token_uri = g_nft_examples[g_devs_for_revolution_label]["token_uri"]
+    const realized = isBase64(token_uri);
+    expect(realized).toEqual(true);
+  });
+
+});
+
+
+
+describe("evaluateNft", function () {
+
+  test('#evaluateNft', async () => {
+
+    // const all_nft_examples = [g_lazy_lions_label, g_alpacadabra_label, g_devs_for_revolution_label];
+    const all_nft_examples = [g_lazy_lions_label, g_alpacadabra_label];
+
+    for(let ex_index = 0; ex_index < all_nft_examples.length; ex_index++) {
+      const nft_example = all_nft_examples[ex_index];
+      const token_uri = g_nft_examples[nft_example]["token_uri"]
+
+      // call
+      const out = await evaluateNft(token_uri);
+
+      // verify output - grading
+      {
+        // const expected = "Green";
+        const expected = g_nft_examples[nft_example]["expected_output"]["nft"]
+        const realized = out[0];
+        expect(realized).toEqual(expected);
+      }
+      ;
+      // verify output - uri grading
+      {
+        // const expected = "tokenUri is IPFS link";
+        const expected = g_nft_examples[nft_example]["expected_output"]["uri"]
+        const realized = out[1];
+        // should be an array
+        expect(expected.length).toEqual(realized.length);
+        for (let i = 0; i < expected.length; i++) {
+          expect(expected[i]).toEqual(realized[i]);
+        }
+
+      }
+      ;
+      // verify output - metadata
+      {
+        const expected = g_nft_examples[nft_example]["expected_output"]["metadata"]
+        const realized = out[2];
+
+        // for the keys we have defined check that they align with what is in realized metadat
+        for (const [key, value] of Object.entries(expected)) {
+          expect(value).toEqual(realized[key]);
+        }
+
+      };
```

Comment:
Is wrapping these statements in { } a stylistic thing for grouping ideas in the tests or do they have a functional use?